### PR TITLE
added support for formats for tv4

### PR DESCRIPTION
**Pull request by eneufeld**
Needed change for #339

### CODE REVIEW

File: src/components/services/services.ts [hunk 1]
```diff
@@ -4,7 +4,14 @@ import {PathUtil} from "./pathutil";
 
 // TODO: replace
 let tv4 = require('tv4');
-
+let formats = require('tv4-formats');
+let validator;
+if(tv4!=null){
```
Comment (edgarmueller) [inline src/components/services/services.ts]: TSLint should complain here (missing WS)

**Comment by edgarmueller** [comment]
Just one word on this: according to https://github.com/ebdrup/json-schema-benchmark, tv4 is quite slow in comparison to ajv (which hinders rendering performance) so we might want to think about switching to ajv soon, as it supports formats out-of-the box.
Otherwise, looks good, except some TSLint warnings.

**Comment by eneufeld** [comment]
I'll check the ts lint. 
I would suggest to fix it with the tv4 formats and then do a second refactoring to replace tv4 with ajv

**Comment by edgarmueller** [comment]
I've [opened an issue](https://github.com/eclipsesource/jsonforms/issues/409) for replacing the validator. Otherwise looks good.

**Pull request merged**